tools/jsdoc/jsdoc_extractor/src/utils.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <algorithm>
+
+template<typename T>
+static bool contains(const std::vector<T>& vec, const T& val)
+{
+    auto it = std::find(vec.cbegin(), vec.cend(), val);
+    return it != vec.cend();
+}
+
+static void ltrim(std::string& s)
+{
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](unsigned char ch) {
+        return !std::isspace(ch);
+    }));
+}
+
+static void rtrim(std::string& s)
+{
+    s.erase(std::find_if(s.rbegin(), s.rend(), [](int ch) {
+        return !std::isspace(ch);
+    }).base(), s.end());
+}
+
+static void trim(std::string& s)
+{
+    ltrim(s);
+    rtrim(s);
+}
+
+static void split(const std::string& str, std::vector<std::string>& out, const std::string& delim)
+{
+    std::size_t current, previous = 0;
+    current = str.find(delim);
+    std::size_t delimLen = delim.length();
+
+    while (current != std::string::npos) {
+        out.push_back(str.substr(previous, current - previous));
+        previous = current + delimLen;
+        current = str.find(delim, previous);
+    }
+    out.push_back(str.substr(previous, current - previous));
+}
+
+static bool startsWith(const std::string& str, const std::string& start)
```

Comment:
It seems strange to implement all this in C++ when a higher language like Python would give you it for free.

Reply:
Actually, I agree with this. This seems more a use case for Python. (Unless we eventually want to go down the route of using LLVM to build an AST and operate on that, but that seems way overkill to me.)